rio_tiler/reader.py

```diff
@@ -204,6 +215,32 @@ def part(
                 width = max_size
                 height = math.ceil(width * ratio)
 
+    if buffer:
```

Comment:
This is a totally different approach for applying buffer. 

The main idea is that we first need to know the output `width` and `height` before adding a `buffer`. For `tile` it's easier because we know the output size (256x256) before calling the `part` method. 

The change proposed here assumed there is a linear link between the output size and the vrt size so we first apply the buffer to `height` and `width` then go back and calculate the new `vrt width/height` and then derive the new transform. 

We could also do it as we were doing before:
- get bounds for `part` or `tile`
- find the output resolution.
- calculate the new bounds (with buffer pixel)
- re-call `get_vrt_transform(` with the new bounds and output width/height. 

🤷